packages/react/src/use-action-form/utils.ts

```diff
@@ -460,6 +460,23 @@ export function applyDataMask(opts: {
 
   const readOnlyPaths = getReadOnlyPaths(select, modelApiIdentifier);
 
+  // Check for relationship fields that are entirely read-only, including nested relationships
+  const checkRelationshipFields = (obj: any, prefix = "") => {
```

Comment:
This makes total sense to me, however I still wonder if we should also just make computed fields a special case in the reshaper?

Thoughts?

Reply:
My thought is that we need both -- regardless of computed fields, I don't think we should be trying to submit relationship data when all the selected fields of the relationship are read only.